lib/validates_zipcode/cldr_regex_collection.rb

```diff
@@ -132,123 +134,3686 @@ module CldrRegexpCollection
+      JM: /\A((JM)(DCN|CHR|AKN|DMR|BPD|AAW|CAN|ACE|DEH|CJS|BMY|BTS|CTY|DWD)(0[1-9]|(1|2)[0-9]|3[1-9]))\z/i,
+      KI: /\A(KI)\d{4}\z/i,
+      KN: /\A(KN)\d{4}\z/i,
+      KP: /\A([A-Z\d\s]){3,8}\z/i,
       LI: /\A\d{4}\z/,
-      LV: /\A(LV)?\-?\d{4}\z/,
+      LV: /\A(LV)?\-?\d{4}\z/i,
       LY: /\A\d{5}\z/,
-      LT: /\A(LT-)?\d{5}\z/,
-      LC: /\A([a-zA-Z\d\s]){3,}\z/,
+      LT: /\A(LT-)?\d{5}\z/i,
+      LC: /\A([A-Z\d\s]){3,}\z/i,
       MC: /\A\d{5}\z/,
-      MD: /\A(([a-zA-Z]){2})(|\s)\d{4}\z/,
-      ME: /\A([a-zA-Z\d\s]){3,}\z/,
+      MD: /\A(([A-Z]){2})(|\s)\d{4}\z/i,
+      ME: /\A([A-Z\d\s]){3,}\z/i,
       MH: /\A\d{5}\z/,
-      MR: /\A([a-zA-Z\d\s]){3,}\z/,
-      MM: /\A([a-zA-Z\d\s]){3,}\z/,
-      MW: /\A([a-zA-Z\d\s]){3,}\z/,
+      MR: /\A([A-Z\d\s]){3,}\z/i,
+      MM: /\A([A-Z\d\s]){3,}\z/i,
+      MW: /\A([A-Z\d\s]){3,}\z/i,
       MZ: /\A\d{4}\z/,
       NA: /\A\d{5}\z/,
       NP: /\A\d{6}\z/,
-      NR: /\A([a-zA-Z\d\s]){3,}\z/,
+      NR: /\A([A-Z\d\s]){3,}\z/i,
       PT: /\A\d{4}([\-]\d{3})?\z/,
       PS: /\A\d{3}\z/,
       PA: /\A\d{4}\z/,
       PE: /\A\d{5}\z/,
-      QA: /\A([a-zA-Z\d\s]){3,}\z/,
-      RW: /\A([a-zA-Z\d\s]){3,}\z/,
-      SC: /\A([a-zA-Z\d\s]){3,}\z/,
+      QA: /\A([A-Z\d\s]){3,}\z/i,
+      RW: /\A([A-Z\d\s]){3,}\z/i,
+      SC: /\A([A-Z\d\s]){3,}\z/i,
       SE: /\A\d{3}[ ]?\d{2}\z/,
       SK: /\A[089]\d{2}[ ]?\d{2}\z/,
-      SL: /\A([a-zA-Z\d\s]){3,}\z/,
-      SB: /\A([a-zA-Z\d\s]){3,}\z/,
-      SR: /\A([a-zA-Z\d\s]){3,}\z/,
-      SO: /\A([a-zA-Z\d\s]){3,}\z/,
-      SV: /\A([a-zA-Z\d\s]){3,}\z/,
+      SL: /\A([A-Z\d\s]){3,}\z/i,
+      SB: /\A([A-Z\d\s]){3,}\z/i,
+      SR: /\A([A-Z\d\s]){3,}\z/i,
+      SO: /\A([A-Z\d\s]){3,}\z/i,
+      SV: /\A([A-Z\d\s]){3,}\z/i,
       SD: /\A\d{5}\z/,
-      ST: /\A([a-zA-Z\d\s]){3,}\z/,
-      SY: /\A([a-zA-Z\d\s]){3,}\z/,
+      ST: /\A([A-Z\d\s]){3,}\z/i,
+      SY: /\A([A-Z\d\s]){3,}\z/i,
       SS: /\A\d{5}\z/,
       TD: /\A\d{5}\z/,
-      TG: /\A([a-zA-Z\d\s]){3,}\z/,
-      TO: /\A([a-zA-Z\d\s]){3,}\z/,
-      TZ: /\A([a-zA-Z\d\s]){3,}\z/,
+      TG: /\A([A-Z\d\s]){3,}\z/i,
+      TO: /\A([A-Z\d\s]){3,}\z/i,
+      TZ: /\A([A-Z\d\s]){3,}\z/i,
       TT: /\A\d{6}\z/,
       TW: /\A\d{3}(\d{2})?\z/,
-      UG: /\A([a-zA-Z\d\s]){3,}\z/,
-      UM: /\A([a-zA-Z\d\s]){3,}\z/,
-      UY: /\A([a-zA-Z\d\s]){3,}\z/,
-      VC: /\A(VC)\d{4}\z/,
+      UG: /\A([A-Z\d\s]){3,}\z/i,
+      UM: /\A([A-Z\d\s]){3,}\z/i,
+      UY: /\A([A-Z\d\s]){3,}\z/i,
+      VC: /\A(VC)\d{4}\z/i,
       VI: /\A\d{5}\z/,
-      VG: /\A([a-zA-Z\d\s]){3,}\z/,
+      VG: /\A([A-Z\d\s]){3,}\z/i,
       VN: /\A\d{6}\z/,
-      VU: /\A([a-zA-Z\d\s]){3,}\z/,
-      WS: /\A([a-zA-Z\d\s]){3,}\z/,
-      YE: /\A([a-zA-Z\d\s]){3,}\z/,
-      ZW: /\A([a-zA-Z\d\s]){3,}\z/
+      VU: /\A([A-Z\d\s]){3,}\z/i,
+      WS: /\A([A-Z\d\s]){3,}\z/i,
+      YE: /\A([A-Z\d\s]){3,}\z/i,
+      ZW: /\A([A-Z\d\s]){3,}\z/i
+    }.freeze
+
+    ZIPCODE_VALUES = {
```

Comment:
This is a great addition for testing the lib. We've being adding tests incrementally, country by country, but to have a full list will make cases explicit and the overall maintenance more effective. Thanks!

I would not add this list on this file though. We only require these examples for testing, so the right place would be the test itself or an additional file for the examples. But more importantly, because this file gets required on any app using this lib, with a potential impact in memory footprints for all of them. 

Can you please move this list to the tests?

Reply:
This comment is obsolete now, but leaving it for reference.